packages/drivers/odsp-driver/src/test/epochTestsWithRedemption.spec.ts

```diff
@@ -55,80 +55,94 @@ describe("Tests for Epoch Tracker With Redemption", () => {
             });
-            await mockFetchSingle(
+    
+            // Initial joinSession call will return 404 but after the timeout, the call will be retried and succeed
+            await mockFetchMultiple(
                 async () => epochTracker.fetchAndParseAsJSON("fetchUrl", {}, "joinSession"),
-                async () => notFound({ "x-fluid-epoch": "epoch1" }),
-                "external");
-        } catch (error: any) {
-            success = false;
-            assert.strictEqual(error.errorType, DriverErrorType.fileNotFoundOrAccessDeniedError,
-                "Error should be file not found or access denied error");
-        }
-        assert.strictEqual(success, false, "Join session should fail if treesLatest call has failed");
+                [
+                    notFound, // joinSession
+                    async () => okResponse({ "x-fluid-epoch": "epoch1" }, {}), // "treesLatest"
+                    async () => okResponse({ "x-fluid-epoch": "epoch1" }, {}), // "joinSession"
+                ],
+            );
+        });
+    
+        it("Requests should fail if joinSession call fails and the getLatest call also fails", async () => {
+            let success: boolean = true;
+    
+            try {
+                epochCallback.setCallback(async () => {
+                    try {
+                        await mockFetchSingle(
+                            async () => epochTracker.fetchAndParseAsJSON("fetchUrl", {}, "treesLatest"),
+                            notFound,
+                            "internal");
+                    } catch (error: any) {
+                        assert.strictEqual(error.errorType, DriverErrorType.fileNotFoundOrAccessDeniedError,
+                            "Error should be file not found or access denied error");
+                    }
+                });
+                await mockFetchSingle(
+                    async () => epochTracker.fetchAndParseAsJSON("fetchUrl", {}, "joinSession"),
+                    async () => notFound({ "x-fluid-epoch": "epoch1" }),
+                    "external");
+            } catch (error: any) {
+                success = false;
+                assert.strictEqual(error.errorType, DriverErrorType.fileNotFoundOrAccessDeniedError,
+                    "Error should be file not found or access denied error");
+            }
+            assert.strictEqual(success, false, "Join session should fail if treesLatest call has failed");
+        });
+    });
+
+    describe("Tests Suite 2", () => {
+        it("Failed treesLatest call should not trigger unhandled rejection event", async () => {
+            const treesLatestP = mockFetchSingle(
+                async () => epochTracker.fetchAndParseAsJSON("fetchUrl", {}, "treesLatest"),
+                notFound,
+            );
+            await assert.rejects(treesLatestP, "should fail without causing an unhandledRejection event.");
```

Comment:
Does this test fail without your change?

Reply:
Yes it fails. Like we see a unhandled rejection.